lib/classifier-reborn/bayes.rb

```diff
@@ -55,19 +59,20 @@ def train(category, text)
       category = CategoryNamer.prepare_name(category)
 
       # Add the category dynamically or raise an error
-      unless @categories.key?(category)
+      unless category_keys.include?(category)
```

Comment:
`@categories` was a nested hash which has `key?` method. However, `category_keys` is an array of just the keys of that hash (which was abstracted to support Redis backend) and arrays don't have `key?` method defined. Am I missing something?